### DISCUSSION TITLE
1:1 call with the ui library

### DISCUSSION THREAD

**Discussion by aslan50** [Azure/communication-ui-library#3194] [General]
https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/voice-video-calling/get-started-with-voice-video-calling-custom-teams-client

Hello,

I utilized this JavaScript example(link above) to establish a user interface (UI) for conducting 1:1 video calls within Teams. In this scenario, an access token and the callee's Teams ID are required.

Is there an existing example that employs the UI Library specifically for this use case? Can I utilize a call composite for achieving this functionality? So far, the examples I've come across only involve providing a Meeting URL or a Group ID, without any specific reference to 1:1 calls. It's possible that there may be future plans to address this, but currently, I haven't found any resources related to it.

Thanks in advice

**Comment by alkwa-msft** [comment]
Can you share more about your scenario?
- I am trying to understand what we mean by "within Teams"
- In this scenario are you trying to do a 1:1 call is it to an ACS user or a Teams user? 

For 1:1/1:n calls we have an offering in public preview
https://azure.github.io/communication-ui-library/?path=/docs/quickstarts-1ton--page

**Reply by aslan50** [reply]
Hi, 

Thanks for your reply. Im trying to call teams User as a Teams user.

**Reply by alkwa-msft** [reply]
I am curious if this doc might be able to help you out. At this time we have not stabilized the feature but it is available in our beta package if you want to try it out.

https://azure.github.io/communication-ui-library/?path=/docs/communicationasteamsuser--page

npm i @azure/communication-react@1.5.1-beta.5

You will need to get a teams token with the correct scope.
https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/voice-video-calling/get-started-with-voice-video-calling-custom-teams-client

You can also try out this scenario with this "preview" on our storybook as well.
https://azure.github.io/communication-ui-library/?path=/story/composites-call-joinexistingcall-asteamsuser--join-existing-call-as-teams-user

**Reply by aslan50** [reply]
I see there is no support yet for this scenario by the Call Composite Component:

https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/voice-video-calling/get-started-with-voice-video-calling-custom-teams-client

I need the functionality like in the link above to render with the UI Components.  A Teams user calls another Teams User with the ObjectId of the User/s.

**Reply by alkwa-msft** [reply]
@aslan50  I can work with my sibling team to understand more about the ETA of when we can expect being able to call another teams user using their ObjectId. The actual UI Library part is plumbed through and we are waiting on the underlying feature to be ready for production loads.

**Reply by alkwa-msft** [reply]
@aslan50  for having the feature in stable we are looking at having it available in the underlying sdk in CY23Q4/CY24Q1. In the mean time, are you able to try out the Call Composite in our beta package for now? That functionality is available there if you want to try it out?
https://www.npmjs.com/package/@azure/communication-react/v/1.7.0-beta.1

**Reply by aslan50** [reply]
@alkwa-msft  thank you for your reply. 

I tested the functionality like desribed in this example:

https://azure.github.io/communication-ui-library/?path=/docs/adhoc-calling--page

However, I had to make a change in the adapter from 'useAzureCommunicationCallAdapter(callAdapterArgs)' to 'useTeamsCallAdapter(callAdapterArgs)' to make it work for calling a Teams user as a Teams user using the CallComposite. It might be helpful to include this example in the Storybook for other users.

But I encountered an issue when attempting to call multiple users in a 1:N (one-to-many) scenario. Is it currently possible to initiate a 1:N call with the composite, or is it something that is being planned for the future?

By calling as an ACS User, you can put a list of ACS User IDs to call in an array, and it starts a group call with the given IDs in the array.

**Comment by alkwa-msft** [comment]
Creating an issue out of this discussion.
I can share this issue link with the underlying calling sdk team that is responsible for this underlying functionality.
https://github.com/Azure/communication-ui-library/issues/3344
